cmd/analyzetestresults/analyzetestresults.go

```diff
@@ -0,0 +1,221 @@
+package analyzetestresults
+
+import (
+	"encoding/xml"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/bsm/ginkgo/v2/reporters"
+	"github.com/konflux-ci/qe-tools/pkg/oci"
+	"k8s.io/klog/v2"
+
+	"github.com/konflux-ci/qe-tools/pkg/types"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// FailureType collects the types of failures met in the PipelineRun
+type FailureType string
+
+const (
+	dropdownSummaryString = "Click to view logs"
+
+	// OtherFailure represents the type failure that hasn't been identified by the analyzer
+	OtherFailure FailureType = "otherFailure"
+	// ClusterCreationFailure represents the issue with provisioning a test cluster
+	ClusterCreationFailure FailureType = "clusterCreationFailure"
+	// TestRunFailure represents the issue with provisioning a test cluster
+	TestRunFailure FailureType = "testRunFailure"
+	// TestCaseFailure represents the issue with provisioning a test cluster
+	TestCaseFailure FailureType = "testCaseFailure"
+)
+
+var (
+	ociArtifactRef              string
+	clusterProvisionLogFilename string
+	jUnitFilename               string
+	e2eTestRunLogFilename       string
+	outputFilename              string
+)
+
+// FailedTestCasesReport collects the data about failures
+type FailedTestCasesReport struct {
+	headerString        string
+	failedTestCaseNames []string
+
+	clusterProvisionLog string
+	e2eTestLog          string
+
+	failureType FailureType
+}
+
+// getTestSuitesFromXMLFile returns all the JUnitTestSuites
+// present within a file with the given name
+func getTestSuitesFromXMLFile(scanner *oci.ArtifactScanner, filename string) (*reporters.JUnitTestSuites, error) {
+	overallJUnitSuites := &reporters.JUnitTestSuites{}
+
+	for _, file := range scanner.FilesPathMap {
+		if string(file.Filename) == filename {
+			if err := xml.Unmarshal([]byte(file.Content), overallJUnitSuites); err != nil {
+				klog.Warningf("cannot decode JUnit suite into xml: %+v", err)
+				return &reporters.JUnitTestSuites{}, err
+			}
+			return overallJUnitSuites, nil
+		}
+	}
+
+	return &reporters.JUnitTestSuites{}, fmt.Errorf("couldn't find the %s file", filename)
+}
+
+// setHeaderString initialises struct FailedTestCasesReport's
+// 'headerString' field based on phase at which PipelineRun failed
+func (f *FailedTestCasesReport) setHeaderString(overallJUnitSuites *reporters.JUnitTestSuites) {
```

Comment:
I'd like to move some of this `if` logic into that report struct and let report struct determine and set the failure type on the struct than this function. 

Then all this function has to do is be more of output formatter. It checks the f.failuretype to set the header string rather than doing both.